py/obj.h

```diff
@@ -604,9 +604,16 @@ extern const mp_obj_type_t mp_type_ZeroDivisionError;
 
 // Constant objects, globally accessible
 // The macros are for convenience only
+#if MICROPY_SMALL_DATA_SECTION
+#define mp_const_none (MP_OBJ_FROM_PTR((void*)MP_SMALL_DATA_CONST_NONE))
+#define mp_const_false (MP_OBJ_FROM_PTR((void*)MP_SMALL_DATA_CONST_FALSE))
+#define mp_const_true (MP_OBJ_FROM_PTR((void*)MP_SMALL_DATA_CONST_TRUE))
```

Comment:
I guess this is necessary because the compiler does not know in advance that these pointers will be at small addresses, so will always generate instructions to access them at large locations (even though the linker eventually knows where they are).  So this forcing of the address is required for it to reduce code size (it'd still work using the original ptrs below, but wouldn't change code size).

Reply:
Yep. You're right here. I'd love a linker that could apply such changes in link-time...

Thinking of it, won't a modern GCC with LTO enabled handle it well? I'll give it a try.

Reply:
Tested with GCC 9.2.1 with LTO - `ldr [pc]` instructions are still emitted :(